### DISCUSSION TITLE
Sugestion -- New App: SOS-SMS

### DISCUSSION THREAD

**Discussion by lianergoist** [FossifyOrg/General-Discussion#164] [Ideas]
This is an app I would love to have, and I am sure many people will find it useful.

Basically, it is an app where you can enter a number and a message, and then save it. Later you can select the message from a list (main screen of the app) and send it with one click. The message will/can also include your (current and/or last known) position.

As the name suggest, it is meant to call for help and give your position in one click. But, it can also be used to say "I'm okay" in one click, for people who are busy doing other things (fx. sailing a boat in stormy weather).

I wrote something similar many years ago (back in the android 4 days), but I have not done any android programming since, so I hope someone else will take up the idea.

It should be a very simple app to write. A list, a bit of database access, get location, send sms. I think that's it.

**Comment by Aga-C** [comment]
I think it wouldn't be doable - only one app at a time can have permission to fetch/send SMS messages. It would be more convenient to have such feature in a normal SMS app.

**Reply by lianergoist** [reply]
> only one app at a time can have permission to fetch/send SMS messages

Are you sure about that? I think you are right about fetching, but also sending?

Well, I did suggest to include it [in #142](https://github.com/FossifyOrg/Messages/issues/142).

**Reply by Aga-C** [reply]
I did a quick lookup, and you're right, `SEND_SMS` can be added to more apps, and Google Play allows it for SOS apps. However, I think it would be better to have it in an SMS app, so that's good you've already created that feature request.

**Comment by PanderMusubi** [comment]
This is also the reason why the app https://github.com/Seva-coder/Finder stopped. However, it might be a good reason to integrate *find my phone / find me / send SOS* functionality into the Messages app. For me, this is default functionality that should be easily available to all.

That the `SEND_SMS` permission leaves no alternative might even be a good thing. Many people that lose their phone or are in an emergency don't have an app for that installed.

Two other apps that offer functionality like this are:
- https://github.com/keepalivedev/KeepAlive
- https://gitlab.com/Nulide/findmydevice

Perhaps the authors and contributors of the three above-mentioned apps might be interested in migrating their work to the Messenger app as it would reach more potential users and maintainers/contributors.

**Comment by naveensingh** [comment]
Closing for now, maybe someday, see also: https://github.com/orgs/FossifyOrg/discussions/166#discussioncomment-9417721
